### PR TITLE
libratbag: 0.17 -> 0.18; piper: 0.7 -> 0.8; format with nixfmt

### PR DESCRIPTION
**Pull request by Gliczy**
## Description of changes
Fixes https://github.com/NixOS/nixpkgs/issues/344886 and https://github.com/NixOS/nixpkgs/issues/344887.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/libratbag/default.nix [hunk 1]
```diff
@@ -1,36 +1,72 @@
...
+  meta = {
     description = "Configuration library for gaming mice";
-    homepage    = "https://github.com/libratbag/libratbag";
-    license     = licenses.mit;
-    maintainers = with maintainers; [ mvnetbiz ];
-    platforms   = platforms.linux;
+    homepage = "https://github.com/libratbag/libratbag";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ mvnetbiz ];
+    platforms = lib.platforms.linux;
```
Comment (Atemu) [inline pkgs/os-specific/linux/libratbag/default.nix:70]: Please separate any refactors from the auto format.

File: pkgs/os-specific/linux/libratbag/default.nix [hunk 2]
```diff
@@ -1,36 +1,72 @@
...
 
   src = fetchFromGitHub {
-    owner  = "libratbag";
-    repo   = "libratbag";
-    rev    = "v${version}";
-    sha256 = "sha256-dAWKDF5hegvKhUZ4JW2J/P9uSs4xNrZLNinhAff6NSc=";
+    owner = "libratbag";
+    repo = "libratbag";
+    rev = "v${version}";
+    hash = "sha256-dAWKDF5hegvKhUZ4JW2J/P9uSs4xNrZLNinhAff6NSc=";
```
Comment (Atemu) [inline pkgs/os-specific/linux/libratbag/default.nix:31]: Same here.

File: pkgs/os-specific/linux/piper/default.nix [hunk 3]
```diff
@@ -30,12 +63,12 @@ python3.pkgs.buildPythonApplication rec {
...
     description = "GTK frontend for ratbagd mouse config daemon";
     mainProgram = "piper";
-    homepage    = "https://github.com/libratbag/piper";
-    license     = licenses.gpl2Only;
-    maintainers = with maintainers; [ mvnetbiz ];
-    platforms   = platforms.linux;
+    homepage = "https://github.com/libratbag/piper";
+    license = lib.licenses.gpl2Only;
+    maintainers = with lib.maintainers; [ mvnetbiz ];
+    platforms = lib.platforms.linux;
```
Comment (Atemu) [inline pkgs/os-specific/linux/piper/default.nix:72]: And here.

**Review by Atemu — APPROVED** [review]
Diff LGTM apart from these minor nits. I'll give it a spin on my desktop machine with a supported mouse.

File: pkgs/os-specific/linux/libratbag/default.nix (hunk 1 above)
Comment (Gliczy) [reply pkgs/os-specific/linux/libratbag/default.nix:70]: Would putting both the `with lib` and `hash` changes into a single commit like `piper: refactor` be fine?

Comment (Atemu) [reply pkgs/os-specific/linux/libratbag/default.nix:70]: Sure.

**Comment by Gliczy** [comment]
Note that `piper` needs `sevices.ratbagd.enable = true` (which adds `libratbag` to `services.dbus.packages`), so I had to manually override `sevices.ratbagd.package` to use it's updated version in order to test.

**Comment by Gliczy** [comment]
Rebased to resolve the merge conflict.

**Review by misuzu — APPROVED** [review]

**Pull request merged**